File: Common/src/main/java/me/mvp/frame/utils/GsonUtils.java
package me.mvp.frame.utils;

import com.google.gson.Gson;

import me.mvp.frame.di.component.AppComponent;
import me.mvp.frame.utils.base.BaseUtils;

/**
 * GsonHelper
 */
public class GsonUtils extends BaseUtils {

    private static Gson gson;

    public GsonUtils() {
        super();
    }

    public static Gson getGson() {
        if (gson == null) {
            gson = new Gson();
        }
        return gson;
    }

    public static Gson getGson(AppComponent component) {
        return component.getGson();
    }

    /**
     * JsonString to Entity
     *
     * @param value
     * @param cls
     */
    public static <T> T getEntity(String value, Class<T> cls) {
        T t = null;
        try {
            t = getGson().fromJson(value, cls);
        } catch (Exception e) {
            e.printStackTrace();
        }

        return t;
    }

    /**
     * Object to JsonString
     *
     * @param src
     */
    public static String getString(Object src) {
        String value = null;
        try {
            value = getGson().toJson(src);
        } catch (Exception e) {
            e.printStackTrace();
        }

        return value;
    }
}

File: Common/src/main/java/me/mvp/frame/utils/GsonUtils.kt
package me.mvp.frame.utils

import com.google.gson.Gson

import me.mvp.frame.di.component.AppComponent
import me.mvp.frame.utils.base.BaseUtils

/**
 * Gson 工具类
 */
class GsonUtils : BaseUtils() {

    companion object {

        private var gson: Gson = Gson()

        fun getGson(component: AppComponent): Gson {
            return component.gson
        }

        /**
         * JsonString to Entity
         *
         * @param value
         * @param cls
         */
        fun <T> getEntity(value: String, cls: Class<T>): T? {
            var t: T? = null
            try {
                t = gson.fromJson(value, cls)
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return t
        }

        /**
         * Object to JsonString
         *
         * @param src
         */
        fun getString(src: Any): String? {
            var value: String? = null
            try {
                value = gson.toJson(src)
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return value
        }
    }
}